core/dockerutil/port_util.go

```diff
@@ -0,0 +1,22 @@
+package dockerutil
+
+import (
+	"github.com/ory/dockertest/v3"
+	dc "github.com/ory/dockertest/v3/docker"
+)
+
+// GetPort returns the port of a container.
+// unlike resource.GetPort this returns the ports in reverse order.
+// this can be useful in ci environments(?)
+func GetPort(resource *dockertest.Resource, port string) string {
```

Comment:
Let's add a bit more explanation here. Tl;dr is before we were returning a port on 0.0.0.0 something docker does not force to be unique, now we just return the last port to get around the 0.0.0.0 thing on non docker-for-mac stuff (you guys have as much context as I do on why this works.  Just ran docker ps and noticed ports were different sometimes, perhaps chatgpt can explain?) if not just explain the behavior  int the comment and say we're not sure why it works

Reply:
Side note. Can we please implement this for explorer/localmetrics in this PR? Just anywhere we're using resource.GetPort:

<img width="889" alt="image" src="https://github.com/synapsecns/sanguine/assets/83933037/14dd9054-df12-470f-9793-9ecc3fd7fd33">


I have a feeling this is also causing explorer flakes (Cc @nautsimon )

Reply:
>Side note. Can we please implement this for explorer/localmetrics in this PR? Just anywhere we're using resource.GetPort

Addressed [here](https://github.com/synapsecns/sanguine/pull/1280/commits/3a1dcb84d39e121ae8ea0bab90af56ff399026f8)

Reply:
> Let's add a bit more explanation here.

Addressed [here](https://github.com/synapsecns/sanguine/pull/1280/commits/bbd958f2f3e39e9b81fc8857569fc248c766dda9)